homeassistant/components/shelly/__init__.py

```diff
@@ -504,13 +508,22 @@ def __init__(self, hass: HomeAssistant, device: BlockDevice) -> None:
             update_interval=timedelta(seconds=update_interval),
         )
         self.device = device
+        self.entry = entry
 
     async def _async_update_data(self) -> None:
         """Fetch data."""
         try:
             async with async_timeout.timeout(AIOSHELLY_DEVICE_TIMEOUT_SEC):
                 _LOGGER.debug("REST update for %s", self.name)
                 await self.device.update_status()
+
+                if self.device.status["uptime"] < 2 * REST_SENSORS_UPDATE_INTERVAL:
+                    old_firmware = self.device.firmware_version
+                    await self.device.update_shelly()
+                    if self.device.firmware_version != old_firmware:
+                        await async_device_update_info(
+                            self.hass, self.device, self.entry
+                        )
```

Comment:
Can be simplified using guard clause:
```suggestion
                if self.device.status["uptime"] > 2 * REST_SENSORS_UPDATE_INTERVAL:
                    return
                old_firmware = self.device.firmware_version
                await self.device.update_shelly()
                if self.device.firmware_version == old_firmware:
                    return
                await async_device_update_info(self.hass, self.device, self.entry)
```